src/internal/operators/last.ts

```diff
@@ -1,26 +1,26 @@
+/** @prettier */
 import { Observable } from '../Observable';
 import { EmptyError } from '../util/EmptyError';
-import { OperatorFunction } from '../../internal/types';
+import { OperatorFunction, TruthyTypesOf } from '../types';
 import { filter } from './filter';
 import { takeLast } from './takeLast';
 import { throwIfEmpty } from './throwIfEmpty';
 import { defaultIfEmpty } from './defaultIfEmpty';
 import { identity } from '../util/identity';
 
-/* tslint:disable:max-line-length */
-export function last<T, D = T>(
-  predicate?: null,
-  defaultValue?: D
-): OperatorFunction<T, T | D>;
+export function last<T>(predicate: BooleanConstructor): OperatorFunction<T, TruthyTypesOf<T>>;
+export function last<T, D>(predicate: BooleanConstructor, defaultValue: D): OperatorFunction<T, TruthyTypesOf<T> | D>;
+export function last<T, D = T>(predicate?: null, defaultValue?: D): OperatorFunction<T, T | D>;
```

Comment:
What about?

```suggestion
export function last<T, D = TruthyTypesOf<T>>(predicate?: null, defaultValue?: D): OperatorFunction<T, TruthyTypesOf<T> | D>;
```

I'm not sure, though :thinking:

Reply:
`last` without a predicate would just give you the last value (or the default if it's empty). And the last value could be `false` or `''`.

Reply:
Ups sorry! I misread the implementation... I thought that it was doing `filter(identity)` when the predicate was not provided (which wouldn't make any sense, ofc) and it's actually doing just `identity`. Sorry about that!